demos/demos_by_use_case/bio/SingleCellDemo.ipynb

```diff
@@ -0,0 +1,935 @@
+        },
+        "id": "YW2M1hwTekeW",
+        "outputId": "7661f1d1-5747-4b9a-ae13-5e3754e6f1cf"
+      },
+      "id": "YW2M1hwTekeW",
+      "execution_count": 21,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "WARNING:graphistry.umap_utils:* Ignoring target column of shape (96, 0) in UMAP fit, as it is not one dimensional"
+          ]
+        },
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "\n",
+            " Total  26.6 seconds passed\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "GPU UMAP"
+      ],
+      "metadata": {
+        "id": "tnH6euKnenSJ"
+      },
+      "id": "tnH6euKnenSJ"
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "EE=pd.DataFrame(adata.X[:,I],columns=adata.uns['gcsGenesNames'][I],index=adata.uns['allcellnames'])\n",
+        "g1=graphistry.nodes(cudf.from_pandas(EE.T)) #,columns=adata1.uns['gcsGenesNames']))\n",
+        "\n",
+        "t0 = time.time()\n",
+        "\n",
+        "g11 = g1.umap(\n",
+        "            use_scaler='robust', ## zscale, minmax, standard, normal,\n",
+        "            n_components=2,\n",
+        "            n_neighbors=12,\n",
+        "            engine='cuml'\n",
+        "    )\n",
+        "\n",
+        "\n",
+        "print('\\n Total ', np.round(time.time() - t0,1), 'seconds passed')\n"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "0X3aqsRCO9gR",
+        "outputId": "22d64c62-8f9f-4499-a8e7-cc5f098bcddd"
+      },
+      "id": "0X3aqsRCO9gR",
+      "execution_count": 19,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "WARNING:graphistry.util.memoization:! Failed umap speedup attempt. Continuing without memoization speedups.WARNING:graphistry.umap_utils:* Ignoring target column of shape (96, 0) in UMAP fit, as it is not one dimensional"
+          ]
+        },
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "\n",
+            " Total  15.5 seconds passed\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "emb2=g11._node_embedding\n",
+        "\n",
+        "A=emb2.reset_index()['index']\n",
+        "emb2.index=A\n",
+        "\n",
+        "B=g11._edges\n",
+        "B['_src_implicit'] = B['_src_implicit'].replace(A, regex=True)\n",
+        "B['_dst_implicit'] = B['_dst_implicit'].replace(A, regex=True)\n",
+        "\n",
+        "\n",
+        "g22=graphistry.nodes(emb2.reset_index(),'index').edges(g11._edges.dropna(),'_src_implicit','_dst_implicit').bind(point_x=\"x\",point_y=\"y\").settings(url_params={\"play\":0})\n",
```

Comment:
can this be just

```python

### Visualize

* Nodes are cells
* Edges are similarity relationships
* Initial layout is from the UMAP dimensionality reduction to 2D
* Interactive layout is an aesthetically-optimized force-directed graph over the similarity graph, which is more interpretable for dense clusters

g11.plot()

```

Note that `g.umap()` already sets up default settings for plotting